assignment/impl/src/java/org/sakaiproject/assignment/persistence/AssignmentRepositoryImpl.java

```diff
@@ -160,6 +160,7 @@ public long countSubmittedSubmissionsForAssignment(String assignmentId) {
                 .add(Restrictions.eq("assignment.id", assignmentId))
                 .add(Restrictions.eq("submitted", Boolean.TRUE))
                 .add(Restrictions.eq("userSubmission", Boolean.TRUE))
+                .add(Restrictions.isNotNull("dateSubmitted"))
```

Comment:
Would be nice if its marked as submitted then dateSubmitted shouldn't be null, however this was different issue that was created when they decided to create submissions for everyone by default.

Reply:
But for those the submitted date is not filled, is it? Like I said on my other comment, just took what was working on 11.x to solve the issues. And I think this way all cases (previous and current) are considered.

Reply:
Yeah my comment here was more about the preferred approach vs the actual! I am fine with this, it was just a comment!